clients/src/test/java/org/apache/kafka/clients/consumer/internals/ConsumerCoordinatorTest.java

```diff
@@ -668,6 +695,7 @@ public void onFailure(RuntimeException e, RequestFuture<Object> future) {
         coordinator.markCoordinatorUnknown("test cause");
         consumerClient.pollNoWakeup();
         assertTrue(asyncCallbackInvoked.get());
+        assertEquals(coordinator.inFlightAsyncCommits.get(), 0);
```

Comment:
I was wondering if we could add this to the AfterEach, but it seems like `testCommitAsyncWithUserAssignedType` would fail.

Reply:
You mean add assertion `assertEquals(coordinator.inFlightAsyncCommits.get(), 0);` in the AfterEach? Hmm, how could that fail that test? Test `testCommitAsyncWithUserAssignedType` already ends with that assertion.

Reply:
I see that you tested this in combination with the comment from next thread. And indeed, I see the same thing.